### PR TITLE
Add addon option to cppcheck

### LINKED ISSUE #1107: [Cppcheck] Support `addon` option
Hi, I want to use misra.py cppcheck addon.
Could you tell me how can I use the addon?

Ref: http://cppcheck.sourceforge.net/manual.pdf
```
9.2 Running Addons
Addons could be run through Cppcheck command line utility as follows:
cppcheck --addon=misra.py somefile.c
This will launch all Cppcheck checks and additionally calls specific checks
provided by selected addon.
Some addons need extra arguments. You can configure how you want to execute
an addon in a json file. For example put this in misra.json:
{
"script": "misra.py",
"args": [
"--rule-texts=misra.txt",
"--suppress-rules 17.3,21.12"
]
}
And then the configuration can be executed on the cppcheck command line:
cppcheck --addon=misra.json somefile.c
By default Cppcheck would search addon at standard path which was specified
in installation process. You also can set this path directly, for example:
cppcheck --addon=/opt/cppcheck/configurations/my_misra.json somefile.c
This allows you create and manage multiple configuration files for different
projects.
```

I see https://help.sider.review//tools/cplusplus/cppcheck#cppcheck
I Could not find about addons option.

### PR DESCRIPTION
**Pull request by dev-yakuza**
Fix #1107 

I added `addon` option for cppcheck.([Cppcheck addons](https://github.com/danmar/cppcheck/tree/master/addons))
You can set `sider.yml` like below to use it.

```yml
linter:
  cppcheck:
    addon: y2038
```

or

```yml
linter:
  cppcheck:
    addon:
      - y2038
      - cert
      - misra
```

The test codes(`y2038-inc.h `, `y2038-test.c `) are from [Cppcheck addons test](https://github.com/danmar/cppcheck/tree/master/addons/test)

### CODE REVIEW

**Comment by ybiquitous** [comment]
Also, please update the doc: https://help.sider.review/tools/cplusplus/cppcheck

**Comment by ybiquitous** [comment]
Also, please update the changelog: https://github.com/sider/runners/blob/f742f47ebb0de3be4626f68ad1bd888405cfefe5/CHANGELOG.md#unreleased

File: images/cppcheck/Dockerfile [hunk 1]
```diff
@@ -20,11 +20,6 @@ RUN cd /tmp && \
       CXXFLAGS="-O2 -DNDEBUG -Wall -Wno-sign-compare -Wno-unused-function" && \
     rm -rf cppcheck-${CPPCHECK_VERSION}
 
-FROM sider/devon_rex_base:2.19.0
```
Comment (yykamei) [inline images/cppcheck/Dockerfile:23]: I'm curious why you decided not to use [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).

Comment (ybiquitous) [reply images/cppcheck/Dockerfile:23]: Because Python runtime is required to run `--addon={...}.py`.

Comment (yykamei) [reply images/cppcheck/Dockerfile:23]: OK, I got it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ All notable changes to this project will be documented in this file.
 
 - Fix "No space left" error on Git workspace [#1112](https://github.com/sider/runners/pull/1112)
 - **GolangCI-Lint** Explicitly set `--timeout` [#1117](https://github.com/sider/runners/pull/1117)
+- Add `addon` option to **cppcheck** [#1119](https://github.com/sider/runners/pull/1119)
```
Comment (ybiquitous) [inline CHANGELOG.md:11]: [suggestion] Same format as other entries:

```suggestion
- **Cppcheck** Add `addon` option [#1119](https://github.com/sider/runners/pull/1119)
```

File: test/smokes/cppcheck/expectations.rb [hunk 3]
```diff
@@ -413,3 +413,155 @@
...
+    {
+      id: "y2038-unsafe-call",
+      path: "y2038-inc.h",
+      location: { start_line: 27 },
+      message: "timespec is Y2038-unsafe",
+      links: [],
+      object: { severity: "warning", verbose: nil, inconclusive: false, cwe: nil, location_info: nil },
+      git_blame_info: nil
+    }
+  ],
```
Comment (ybiquitous) [inline test/smokes/cppcheck/expectations.rb:565]: [question] Can we reduce similar cases to make the test code readable?

Comment (dev-yakuza) [reply test/smokes/cppcheck/expectations.rb:565]: I will change the test code to reduce same errors.

Thanks

Comment (ybiquitous) [reply test/smokes/cppcheck/expectations.rb:448]: Great. 👍

**Comment by dev-yakuza** [comment]
Thanks for your review 🙇‍♂️

**Pull request merged**